pkg/kubelet/util/util_windows.go

```diff
@@ -123,3 +131,37 @@ func GetBootTime() (time.Time, error) {
 	}
 	return currentTime.Add(-time.Duration(output) * time.Millisecond).Truncate(time.Second), nil
 }
+
+type reparseDataBufferHeader struct {
+	ReparseTag        uint32
+	ReparseDataLength uint16
+	Reserved          uint16
+}
+
+type reparseDataBuffer struct {
+	Header reparseDataBufferHeader
+	Detail [syscall.MAXIMUM_REPARSE_DATA_BUFFER_SIZE]byte
+}
+
+// IsUnixDomainSocket returns whether a given file is a AF_UNIX socket file
+func IsUnixDomainSocket(fileInfo os.FileInfo, filePath string) (bool, error) {
+	fd, err := windows.CreateFile(windows.StringToUTF16Ptr(filePath), windows.GENERIC_READ, 0, nil, windows.OPEN_EXISTING, windows.FILE_FLAG_OPEN_REPARSE_POINT|windows.FILE_FLAG_BACKUP_SEMANTICS, 0)
```

Comment:
Consider adding a comment and/or a pointer to golang/go#33357 here. I'm not sure what "reparse point" is/means and the logic below is a bit cryptic.